fix: generate a non-ascii binary signature

enriquebelarte opened the pull request:
GPG signature needed should be a non-ASCII binary file as generated in a `gpg --detach-sign` command. Now we are getting a clear ASCII GPG signature file.

More context:
https://issues.redhat.com/browse/ISV-4394
https://developer.hashicorp.com/terraform/registry/providers/publishing#manually-preparing-a-release

/cc @ernesgonzalez33 @ralphbean

mmalina commented on tasks/sign-base64-blob/README.md:
```diff
@@ -28,3 +28,5 @@ data:
 
 ## Changes in 1.0.1
 - Now the task decodes the payload from base64 before creating the `.sig` file
+## Changes in 1.0.2
```
We typically put the newest to the top. Can you move this above 1.0.1 and add an empty line in between?

enriquebelarte commented:
/hold

mmalina commented on tasks/sign-base64-blob/README.md:
```diff
@@ -28,3 +28,5 @@ data:
 ## Changes in 1.0.1
 - Now the task decodes the payload from base64 before creating the `.sig` file
+## Changes in 1.0.2
+- The `.sig` file is not a clear ASCII GPG signature anymore but a not ASCII armored GPG binary.
```
It's not clear to me what's happening. At first I thought the payload is non-ascii and we want to make ascii from it. But it's the opposite, right? Just curious, why is this change happening? In any case, I would suggest to reword this slightly to make it a bit clearer, e.g.:
```
- Save the `.sig` file as a non-ASCII armored GPG binary instead of clear ASCII GPG signature
```

enriquebelarte replied:
Right from the outset, our objective was to generate a GPG signature in non-ASCII binary format in accordance with the specified requirement. However, upon testing the whole workflow, we saw that the resulting GPG .sig file is in clear ASCII format, contrary to the intended. Hence the PR. Will reword with your suggestion.

mmalina replied:
OK, thanks for the explanation.

enriquebelarte commented:
/unhold

mmalina commented:
> This branch is out-of-date with the base branch

@enriquebelarte can you rebase, please?

enriquebelarte committed:
fix: generate a non-ascii binary signature

Save the file as a non-ASCII armored GPG binary instead
 of clear ASCII GPG signature.

Signed-off-by: Enrique Belarte Luque <ebelarte@redhat.com>

johnbieren commented on tasks/sign-base64-blob/tests/mocks.sh:
```diff
@@ -23,3 +23,13 @@ function kubectl() {
+function gpg() {
+  if [ $* != "get internalrequest internal-request -o=jsonpath='{.status.results.signed_payload}'" ]
+  then
+    echo "Unexpected call to kubectl"
```
I think this is a copy paste error

johnbieren commented on tasks/sign-base64-blob/tests/mocks.sh:
```diff
@@ -23,3 +23,13 @@ function kubectl() {
 }
+
+function gpg() {
+  if [ $* != "get internalrequest internal-request -o=jsonpath='{.status.results.signed_payload}'" ]
```
I think this is a copy paste error as gpg is not called with these arguments in the task

enriquebelarte replied:
/hold You are right.

mmalina replied:
It would have been cool if we could produce a real gpg snippet that `gpg --dearmor` could produce without mocking, but it probably cannot easily be done - we would need to set up gpg keys, probably too much hassle. So the mock could just have a `cat` or maybe print something before, that could be checked in the result, like:
```sh
function gpg() {
  if [ $* != "--dearmor" ]; then
    echo "Unexpected call to gpg"
    exit 1
  fi
  echo -n "Dearmored: "
  cat
}
```

enriquebelarte replied:
Sure, setting up GPG keys would be the ideal test and I expect to do it in a future, but it's not something I can focus on right now.

enriquebelarte commented:
/unhold

mmalina reviewed (changes requested):
@enriquebelarte can you change the commit message? It currently says: `fix: add dummy test function to avoid GPG failure`

It should probably say the same as the PR title: `fix: generate a non-ascii binary signature` plus ideally a short description.
